### PR TITLE
464 season home page links

### LINKED ISSUE #464: Access control for season home page links
Register - unless enrolled
Send test task -if enrolled &&  !test_task_done
Interview -  if test_task_done && !interviewing
Hometask - attending && !dropped && !no_home_work_per_this_lecture

### PR DESCRIPTION
**Pull request by nadiyaka**
Resolves #464

Access control for season home page links

### CODE REVIEW

File: components/courses/app/policies/courses/season/send_homework_policy.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      private
+
+      attr_reader :season, :student
+
+      def season_has_status_live?
+        season.status.to_sym == Courses::Season::LIVE
+      end
+
+      def student_is_dropped?
+        student && student.try(:dropped?)
```
Comment (sveredyuk) [inline components/courses/app/policies/courses/season/send_homework_policy.rb]: `student&.try(: dropped?)`

File: components/courses/spec/policies/courses/season/interviews_policy_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+describe Courses::Season::InterviewsPolicy do
+  let(:season)  { create(:season) }
+  let(:student) { create(:student) }
+  let(:policy)  { Courses::Season::InterviewsPolicy.new(season, student) }
+
+  describe '#allowed?' do
+    context "season has status registration, student has done a test task & student doesn't have an interview yet" do
+      it 'allows to pass policy' do
+        policy.stub(:season_has_status_registration?).and_return(true)
```
Comment (sveredyuk) [inline components/courses/spec/policies/courses/season/interviews_policy_spec.rb]: stub it's rspec2 method, imho it's deprecated in rspec, try:
`allow(policy).to receive(: season_has_status_registration?).and_return(true)`

**Review by sveredyuk — APPROVED** [review]
Good job. Just fix stub method

**Comment by sveredyuk** [comment]
and resolve conflicts

File: components/courses/app/policies/courses/season/send_homework_policy.rb (hunk 1 above)
Comment (biow0lf) [reply components/courses/app/policies/courses/season/send_homework_policy.rb]: student&.dropped?

Comment (biow0lf) [reply components/courses/app/policies/courses/season/send_homework_policy.rb]: Просто метод & это переписанный на си и добавленный в синтаксис руби метод try из рельсы.

Comment (nadiyaka) [reply components/courses/app/policies/courses/season/send_homework_policy.rb]: Thanks!

**Review by sveredyuk — APPROVED** [review]

**Pull request merged**